libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx

```diff
@@ -1,193 +1,126 @@
-    (state) => selectApplicationById(state, applicationId),
-    (a, b) => {
-      return JSON.stringify(a?.id) === JSON.stringify(b?.id)
-    }
-  ) as ApplicationEntity | undefined
-
-  const [loading, setLoading] = useState(false)
-
-  const dispatch = useDispatch<AppDispatch>()
-
-  const toasterCallback = () => {
-    if (application) {
-      dispatch(
-        postApplicationActionsRedeploy({
-          applicationId,
-          environmentId,
-          serviceType: getServiceType(application),
-          callback: () =>
-            navigate(
-              ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId) + DEPLOYMENT_LOGS_URL(applicationId)
-            ),
-          queryClient,
-        })
-      )
-    }
-  }
-
-  useEffect(() => {
-    if (application) {
-      methods.setValue('max_duration', application.max_duration_seconds)
-      methods.setValue('nb_restarts', application.max_nb_restart)
-      methods.setValue('port', application.port || undefined)
-
-      // TODO: should be typeguard
-      if (isCronJob(application)) {
-        const { cronjob } = (application as CronJob).schedule
-        methods.setValue('schedule', cronjob?.scheduled_at || undefined)
-        methods.setValue('cmd_arguments', JSON.stringify(cronjob?.arguments) || undefined)
-        methods.setValue('image_entry_point', cronjob?.entrypoint || undefined)
-      } else {
-        const { on_start, on_delete, on_stop } = (application as LifecycleJob).schedule
-        methods.setValue('on_start.enabled', !!on_start)
-        if (on_start?.arguments && on_start?.arguments.length > 0) {
-          methods.setValue('on_start.arguments_string', JSON.stringify(on_start.arguments))
-        }
-        methods.setValue('on_start.entrypoint', on_start?.entrypoint)
-
-        methods.setValue('on_stop.enabled', !!on_stop)
-        if (on_stop?.arguments && on_stop?.arguments.length > 0) {
-          methods.setValue('on_stop.arguments_string', JSON.stringify(on_stop?.arguments))
-        }
-        methods.setValue('on_stop.entrypoint', on_stop?.entrypoint)
-
-        methods.setValue('on_delete.enabled', !!on_delete)
-        if (on_delete?.arguments && on_delete?.arguments.length > 0) {
-          methods.setValue('on_delete.arguments_string', JSON.stringify(on_delete?.arguments))
-        }
-        methods.setValue('on_delete.entrypoint', on_delete?.entrypoint)
+  const { environmentId = '', applicationId = '' } = useParams()
+
+  const { data: service } = useService({ serviceId: applicationId, serviceType: 'JOB' })
+  const { mutate: editService, isLoading: isLoadingEditService } = useEditService({ environmentId })
+
+  const schedule = match(service)
+    .with({ job_type: 'CRON' }, (s) => {
+      const { cronjob } = s.schedule
+      return {
+        schedule: cronjob?.scheduled_at,
+        cmd_arguments:
+          cronjob?.arguments && cronjob?.arguments.length > 0 ? JSON.stringify(cronjob?.arguments) : undefined,
```

Comment:
```suggestion
        cmd_arguments: JSON.stringify(cronjob?.arguments),
```

Reply:
![2024-01-05-120017_230x81_scrot](https://github.com/Qovery/console/assets/1716173/ae109ae9-374e-4a67-b814-30890b26a7c1)

Reply:
same issue line 31, 37 ,43

Reply:
It's why we use eval, you can see the thread bellow